utilities/Common/CounterSimulator.cs

```diff
@@ -193,5 +205,60 @@ public GetMaxQubitCountImpl(CounterSimulator m) : base(m)
             };
         }
         #endregion
+
+        #region Count multi-qubit operations (except Measure)
+        public void CountMultiQubitOperations(ICallable op, IApplyData data)
+        {
+            int nQubits = 0;
+            using (IEnumerator<Qubit> enumerator = data?.Qubits?.GetEnumerator()) {
+                if (enumerator is null) {
+                    // the operation doesn't have qubit parameters
+                    return;
+                }
+                while (enumerator.MoveNext() && nQubits < 2)
```

Comment:
I would do `nQubits = enumerator.Take(2).Count();`.

Reply:
`Take` seems to be a method of LINQ `IEnumerable`, while `enumerator` is `IEnumerator`, and I can't find a neat way to cast `IEnumerator` to `IEnumerable`. I'm not sure the benefits of using `.Take` outweigh the hassle of adding LINQ to the dependencies and casting the enumerables.